core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -269,15 +269,37 @@ void testStartup() {
     void testCopyLogSegmentsToRemoteShouldCopyExpectedLogSegment() throws Exception {
         long oldSegmentStartOffset = 0L;
         long nextSegmentStartOffset = 150L;
-        long oldSegmentEndOffset = nextSegmentStartOffset - 1;
+        long lso = 250L;
+        long leo = 300L;
+        assertCopyExpectedLogSegmentsToRemote(oldSegmentStartOffset, nextSegmentStartOffset, lso, leo);
+    }
+
+    /**
+     * The following values will be equal when the active segment gets rotated to passive and there are no new messages:
+     * last-stable-offset = high-water-mark = log-end-offset = base-offset-of-active-segment.
+     * This test asserts that the active log segment that was rotated after log.roll.ms are copied to remote storage.
+     */
+    @Test
+    void testCopyLogSegmentToRemoteForStaleTopic() throws Exception {
```

Comment:
Good test for the edge case of a stale topic with no activity for longer durations.